bentoml/cli/__init__.py

```diff
@@ -325,18 +325,19 @@ def serve_gunicorn(
 
         if enable_microbatch:
             # avoid load model before gunicorn fork
-            bento_service_metadata_pb = load_bento_service_metadata(
-                bento_service_bundle_path
-            )
             with reserve_free_port() as api_server_port:
-                marshal_server = MarshalServer(
-                    port=port, target_host="localhost", target_port=api_server_port
+                marshal_server = GunicornMarshalServer(
+                    target_host="localhost",
+                    target_port=api_server_port,
+                    bundle_path=bento_service_bundle_path,
+                    port=port,
+                    num_of_workers=1,
```

Comment:
should we make this configurable? so it's easier for us to do experiments in benchmark later